editor/src/core/shared/uid-utils.ts

```diff
@@ -91,7 +94,18 @@ const atoz = [
 export function generateConsistentUID(
   existingIDs: Set<string>,
   possibleStartingValue: string,
+  comments: ParsedComments = emptyComments,
 ): string {
+  const commentFlag = deepFindUtopiaCommentFlag(comments ?? null, 'uid')
+  if (commentFlag != null && isUtopiaCommentFlagUid(commentFlag)) {
+    const { value } = commentFlag
+    COMMENT_FLAG_UIDS.current.add(value)
+    return value
+  }
+  if (COMMENT_FLAG_UIDS.current.has(possibleStartingValue)) {
+    return possibleStartingValue
+  }
+
```

Comment:
is this the right place for this? how do we parse and use the data-uid attribute for elements? I _think_ if an element has a data-uid attribute, we don't even visit the `generateConsistentUID` code path

Reply:
I think that's not the case - that function is used by `getUIDBasedOnElement` which is what we call when generating the jsx conditional expressions